### ISSUE TITLE
problem installing, web ui working but api not

### ISSUE THREAD

**Issue by yosefy** [guardian/alerta#273]
hi i am getting this:
can you help pls?

installed with a script from repo
tryed both ubuntu14, ubuntu 16

[Tue Nov 01 12:04:03.174007 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266] mod_wsgi (pid=25306): Target WSGI script '/var/www/api.wsgi' cannot be loaded as Python module.
[Tue Nov 01 12:04:03.174112 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266] mod_wsgi (pid=25306): Exception occurred processing WSGI script '/var/www/api.wsgi'.
[Tue Nov 01 12:04:03.174280 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266] Traceback (most recent call last):
[Tue Nov 01 12:04:03.174653 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]   File "/var/www/api.wsgi", line 4, in <module>
[Tue Nov 01 12:04:03.174708 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]     from alerta.app import app as application
[Tue Nov 01 12:04:03.174723 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]   File "/opt/alerta/lib/python2.7/site-packages/alerta/app/__init__.py", line 96, in <module>
[Tue Nov 01 12:04:03.174887 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]     db = Mongo()
[Tue Nov 01 12:04:03.174906 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]   File "/opt/alerta/lib/python2.7/site-packages/alerta/app/database.py", line 30, in __init__
[Tue Nov 01 12:04:03.174928 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]     self.connect()
[Tue Nov 01 12:04:03.174970 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]   File "/opt/alerta/lib/python2.7/site-packages/alerta/app/database.py", line 91, in connect
[Tue Nov 01 12:04:03.174994 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]     self._create_indexes()
[Tue Nov 01 12:04:03.175006 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]   File "/opt/alerta/lib/python2.7/site-packages/alerta/app/database.py", line 95, in _create_indexes
[Tue Nov 01 12:04:03.175025 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]     self._db.alerts.create_index([('environment', ASCENDING), ('resource', ASCENDING), ('event', ASCENDING)], unique=True)
[Tue Nov 01 12:04:03.175038 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]   File "/opt/alerta/lib/python2.7/site-packages/pymongo/collection.py", line 1387, in create_index
[Tue Nov 01 12:04:03.175193 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]     self.__create_index(keys, kwargs)
[Tue Nov 01 12:04:03.175213 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]   File "/opt/alerta/lib/python2.7/site-packages/pymongo/collection.py", line 1294, in __create_index
[Tue Nov 01 12:04:03.175262 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]     with self._socket_for_writes() as sock_info:
[Tue Nov 01 12:04:03.175280 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]   File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
[Tue Nov 01 12:04:03.175331 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]     return self.gen.next()
[Tue Nov 01 12:04:03.175373 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]   File "/opt/alerta/lib/python2.7/site-packages/pymongo/mongo_client.py", line 762, in _get_socket
[Tue Nov 01 12:04:03.175416 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]     server = self._get_topology().select_server(selector)
[Tue Nov 01 12:04:03.175433 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]   File "/opt/alerta/lib/python2.7/site-packages/pymongo/topology.py", line 210, in select_server
[Tue Nov 01 12:04:03.175465 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]     address))
[Tue Nov 01 12:04:03.175545 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]   File "/opt/alerta/lib/python2.7/site-packages/pymongo/topology.py", line 186, in select_servers
[Tue Nov 01 12:04:03.175572 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266]     self._error_message(selector))
[Tue Nov 01 12:04:03.175616 2016] [:error] [pid 25306:tid 140499057919744] [remote 192.168.101.9:59266] ServerSelectionTimeoutError: No servers found yet

**Comment by satterly** [comment]
What script? Which repo? What was the exact command? How are you running alerta server? Via commang line or as a WSGI app? If WSGI, are you using apache, nginx, uwsgi, mod_wsgi?

**Comment by yosefy** [comment]
there is a shell script to install in alerta repo which installs everything.so i am using what that script installes.apache with mod i think. script is in root fir of git : install.sh

**Comment by satterly** [comment]
Change the apache config file to this:

```
Listen 8080
<VirtualHost *:8080>
  ServerName localhost
  WSGIDaemonProcess alerta processes=5 threads=5
  WSGIProcessGroup alerta
  WSGIApplicationGroup %{GLOBAL}
  WSGIScriptAlias / /var/www/api.wsgi
  WSGIPassAuthorization On
  <Directory /opt/alerta>
    Require all granted
  </Directory>
</VirtualHost>
<VirtualHost *:80>
  ProxyPass /api http://localhost:8080
  ProxyPassReverse /api http://localhost:8080
  DocumentRoot /var/www/html
  <Directory /var/www/html>
    Require all granted
  </Directory>
</VirtualHost>
EOF
```
Let me know if that fixes it so I can update the `install.sh` script.

**Comment by yosefy** [comment]
it fixed it
thanks :)

**Comment by satterly** [comment]
Fixed by #275
